internal/helmdeployer/helmcache/secret.go

```diff
@@ -0,0 +1,76 @@
+package helmcache
+
+import (
+	"context"
+
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type SecretClient struct {
```

Comment:
If `SecretClient` is meant to implement [Kubernetes' SecretsInterface](https://pkg.go.dev/k8s.io/client-go/kubernetes/typed/core/v1#SecretInterface), maybe it's worth adding a compile-time assertion:
```
var _ v1.SecretInterface = (*SecretClient)(nil)
```

Reply:
I don't see the benefit of adding this. It would fail [here](https://github.com/raulcabello/fleet/blob/helm-cache/internal/helmdeployer/deployer.go#L774) as it is expecting a SecretInterface

Reply:
That's my point, to add this assertion with whatever interface is expected to be implemented from the users.

Reply:
Resolving after reading: https://github.com/rancher/fleet/pull/1809/files#r1338677031